lib/api/collection_config.rb

```diff
@@ -0,0 +1,120 @@
+module Api
+  class CollectionConfig
+    def self.names_for_feature(product_feature_name)
+      names_for_features[product_feature_name]
+    end
+
+    def self.name_for_klass(resource_klass)
+      Settings.collections.detect do |_, spec|
+        spec[:klass] && spec[:klass].constantize == resource_klass
+      end.try(:first)
+    end
+
+    def self.what_refers_to_feature(product_feature_name)
+      referenced_identifiers[product_feature_name]
+    end
+
+    def self.collections_with_description
+      Settings.collections.each_with_object({}) do |(collection, cspec), result|
+        result[collection] = cspec[:description] if cspec[:options].include?(:collection)
+      end
+    end
+
+    def self.find(collection_name)
+      config = Settings.collections[collection_name.to_sym]
+      new(config) if config
```

Comment:
> I still prefer the [] signature you had before, i.e. CollectionConfig[:vms], maybe also support CollectionConfig.vms (when accessing by specific collection names). 

What's your reason for preferring `[]`? I prefer `find` as it's more object oriented. I also prefer `find("vms")` to `CollectionConfig.vms` for the same reason.

> Any ways to not have new(), some methods call CollectionConfig.find several times

I'm not sure I follow. Is your concern that creating an object would be expensive? Or that it would cause unnecessary GCs? IMHO I think it would be better to have a more object oriented solution (which we would only get through newing up objects) unless we could prove that it's detrimental. I would however like to store the main collection somewhere - possibly in the `@req` object?